cdk/src/open-data-platform/lambda/write-demographic-data-handler.ts

```diff
@@ -0,0 +1,233 @@
+// asset-input/src/open-data-platform/lambda/write-demographic-data-handler.js
+import {APIGatewayEvent, APIGatewayProxyCallback, Context} from 'aws-lambda';
+
+const AWS = require('aws-sdk');
+const parse = require('csv-parser');
+const S3 = new AWS.S3();
+const rdsDataService = new AWS.RDSDataService();
+
+const GEO_ID = 'GEOID';
+const RACE_TOTAL = 'RaceTotal';
+const WHITE_POPULATION = 'Estimate!!Total:!!White alone';
+const BLACK_POPULATION = 'Estimate!!Total:!!Black or African American alone';
+
+/// Prepared SQL statements.
+const CREATE_DEMOGRAPHICS_TABLE =
+  'CREATE TABLE IF NOT EXISTS demographics( \n' +
+  '   census_geo_id varchar(255) NOT NULL, \n' +
+  '   total_population real, \n' +
+  '   black_percentage real,\n' +
+  '   white_percentage real, \n' +
+  '   PRIMARY KEY(census_geo_id) \n' +
+  ');';
+
+const DELETE_DEMOGRAPHICS_TABLE =
+  'DELETE FROM demographics WHERE census_geo_id IS NOT NULL';
+
+/// Reads the S3 CSV file and returns [DemographicsTableRow]s.
+const parseS3IntoDemographicsTableRow =
+  (s3Params: Object): Promise<Array<DemographicsTableRow>> => {
+    let results: DemographicsTableRow[] = [];
+    return new Promise(function (resolve, reject) {
+      let count = 0;
+      let fileStream = S3.getObject(s3Params).createReadStream();
+      fileStream.pipe(parse())
+                .on('data',
+                  (chunk) => {
+                    fileStream.pause();
+                    // Only process 10 rows for now.
+                    if (count < 10) {
+                      let row =
+                        new DemographicsTableRowBuilder()
+                          .censusGeoId(chunk[GEO_ID])
+                          .totalPopulation(parseInt(chunk[RACE_TOTAL]))
+                          .blackPopulation(parseInt(chunk[BLACK_POPULATION]))
+                          .whitePopulation(parseInt(chunk[WHITE_POPULATION]))
+                          .build();
+                      count += 1;
+                      console.log(row);
```

Comment:
This log is fine while there are only 10 rows, but will be messy after that. I'd suggest cutting it now.